scripts/quests/i_can_hear_a_rainbow.lua

```diff
@@ -0,0 +1,158 @@
+            [tpz.zone.WESTERN_ALTEPA_DESERT] = true,
+        },
+    },
+    [tpz.weather.NONE] =
+    {
+        bit = 1,
+        zones =
+        {
+            [tpz.zone.WEST_RONFAURE] = true,
+            [tpz.zone.EAST_RONFAURE] = true,
+            [tpz.zone.NORTH_GUSTABERG] = true,
+            [tpz.zone.SOUTH_GUSTABERG] = true,
+            [tpz.zone.EASTERN_ALTEPA_DESERT] = true,
+            [tpz.zone.WEST_SARUTABARUTA] = true,
+            [tpz.zone.EAST_SARUTABARUTA] = true,
+            [tpz.zone.BUBURIMU_PENINSULA] = true,
+            [tpz.zone.YHOATOR_JUNGLE] = true,
+        },
+    },
+    [tpz.weather.DUST_STORM] =
+    {
+        bit = 2,
+        zones =
+        {
+            [tpz.zone.VALKURM_DUNES] = true,
+            [tpz.zone.BATALLIA_DOWNS] = true,
+            [tpz.zone.KONSCHTAT_HIGHLANDS] = true,
+            [tpz.zone.EASTERN_ALTEPA_DESERT] = true,
+            [tpz.zone.TAHRONGI_CANYON] = true,
+            [tpz.zone.MERIPHATAUD_MOUNTAINS] = true,
+            [tpz.zone.SAUROMUGUE_CHAMPAIGN] = true,
+            [tpz.zone.WESTERN_ALTEPA_DESERT] = true,
+        },
+    },
+    [tpz.weather.WIND] =
+    {
+        bit = 3,
+        zones =
+        {
+            [tpz.zone.LA_THEINE_PLATEAU] = true,
+            [tpz.zone.CAPE_TERIGGAN] = true,
+            [tpz.zone.TAHRONGI_CANYON] = true,
+            [tpz.zone.BUBURIMU_PENINSULA] = true,
+        },
+    },
+    [tpz.weather.RAIN] =
+    {
+        bit = 4,
+        zones =
+        {
+            [tpz.zone.LA_THEINE_PLATEAU] = true,
+            [tpz.zone.JUGNER_FOREST] = true,
+            [tpz.zone.PASHHOW_MARSHLANDS] = true,
+            [tpz.zone.ROLANBERRY_FIELDS] = true,
+            [tpz.zone.BUBURIMU_PENINSULA] = true,
+            [tpz.zone.THE_SANCTUARY_OF_ZITAH] = true,
+            [tpz.zone.YUHTUNGA_JUNGLE] = true,
+            [tpz.zone.YHOATOR_JUNGLE] = true,
+        },
+    },
+    [tpz.weather.SNOW] =
+    {
+        bit = 5,
+        zones =
+        {
+            [tpz.zone.BATALLIA_DOWNS] = true,
+            [tpz.zone.BEAUCEDINE_GLACIER] = true,
+            [tpz.zone.XARCABARD] = true,
+        },
+    },
+    [tpz.weather.THUNDER] =
+    {
+        bit = 6,
+        zones =
+        {
+            [tpz.zone.JUGNER_FOREST] = true,
+            [tpz.zone.KONSCHTAT_HIGHLANDS] = true,
+            [tpz.zone.PASHHOW_MARSHLANDS] = true,
+            [tpz.zone.SAUROMUGUE_CHAMPAIGN] = true,
+            [tpz.zone.THE_SANCTUARY_OF_ZITAH] = true,
+        },
+    },
+}
+
+-------------------------------------------------
+-- public functions
+-------------------------------------------------
+
+quests.i_can_hear_a_rainbow.onZoneIn = function(player)
+    if player:hasItem(1125, 0) and player:getQuestStatus(WINDURST, tpz.quest.id.windurst.I_CAN_HEAR_A_RAINBOW) == QUEST_ACCEPTED then
+        local trigger = false
+
+        -- get data for basic weather (e.g. downgrade GALES to WIND)
+        local weather = player:getWeather()
```

Comment:
`entity:getWeather()` will return SCH weather too; are those lost on zoning? Does the effect loss include logging?

Reply:
use zone entities weather instead. sch 'weather' doesn't work for smn quest

Reply:
I don't currently see a way to get zone's weather into LUA, so will have to either add a getWeather binding to lua_zone, and/or add an optional ignoreScholar bool parameter to player:getWeather.  Any preference?

Reply:
when the object type isn't a player or mob, it assumed we want zone weather:
```lua
inline int32 CLuaBaseEntity::getWeather(lua_State *L)
{
    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);

    WEATHER weather = WEATHER_NONE;
    if (m_PBaseEntity->objtype & TYPE_PC || m_PBaseEntity->objtype & TYPE_MOB)
        weather = battleutils::GetWeather((CBattleEntity*)m_PBaseEntity, false);
    else
        weather = zoneutils::GetZone(m_PBaseEntity->getZone())->GetWeather();

    lua_pushinteger(L, weather);

    return 1;
}
```
so you can get the zone (from the player if need be) and then
`zone:getWeather()`
works

Reply:
`player:getZone():getWeather()`

for example

Reply:
Sorry I see what you are saying about it not being in lua_zone...I thought we already had this functionality in use elsewhere already.

Quick search says I am wrong